Hakawai/Mentions/HKWMentionsStartDetectionStateMachine.m

```diff
@@ -202,22 +209,46 @@ - (void)characterTyped:(unichar)c asInsertedCharacter:(BOOL)inserted previousCha
     }
 }
 
-- (void)deleteTypedCharacter:(unichar)deletedChar withCharacterNowPrecedingCursor:(unichar)precedingChar {
+- (void)deleteTypedCharacter:(unichar)deletedChar
+withCharacterNowPrecedingCursor:(unichar)precedingChar
+                    location:(NSUInteger)location
+                textViewText:(NSString *)textViewText {
     // Determine the type of the character
     enum CharacterType {
         CharacterTypeSeparator = 0,
-        CharacterTypeNormal
+        CharacterTypeNormal,
+        CharacterTypeControlCharacter
     };
     enum CharacterType deletedCharacterType = ([HKWMentionsStartDetectionStateMachine.separatorSet characterIsMember:deletedChar] || deletedChar == 0)
     ? CharacterTypeSeparator
     : CharacterTypeNormal;
-    enum CharacterType currentCharacterType = ([HKWMentionsStartDetectionStateMachine.separatorSet characterIsMember:precedingChar] || precedingChar == 0)
-    ? CharacterTypeSeparator
-    : CharacterTypeNormal;
+    __strong __auto_type delegate = self.delegate;
 
+    // Determine the type of the character
+    enum CharacterType currentCharacterType = CharacterTypeNormal;
+    if ([HKWMentionsStartDetectionStateMachine.whitespaceSet characterIsMember:precedingChar] || precedingChar == 0) {
```

Comment:
there should be a function where you pass in a char and ire returns the chartype